File: src_backup/main/java/com/myudog/myulib/api/core/floating/IFloatingObject.java
package com.myudog.myulib.api.core.floating;

import net.minecraft.world.phys.Vec3;
import org.joml.Vector3f;

public interface IFloatingObject {
	void spawn(Vec3 pos);

	void remove();

	void moveTo(Vec3 pos, int interpolationDuration);

	void setScale(Vector3f scale, int interpolationDuration);

	void setRotation(Vector3f leftRotation, int interpolationDuration);
}

File: src/main/kotlin/com/myudog/myulib/api/object/IFloatingObject.kt
package com.myudog.myulib.api.`object`

import net.minecraft.util.math.Vec3d
import org.joml.Vector3f

/**
 * 懸浮物體接口，支援位移、旋轉與縮放
 */
interface IFloatingObject {
    fun spawn(pos: Vec3d)
    fun remove()

    // 平滑變換 (Interpolation)
    fun moveTo(pos: Vec3d, interpolationDuration: Int)
    fun setScale(scale: Vector3f, interpolationDuration: Int)
    fun setRotation(leftRotation: Vector3f, interpolationDuration: Int)
}